munit/shared/src/main/scala/munit/MUnitRunner.scala

```diff
@@ -144,80 +144,215 @@ class MUnitRunner(val cls: Class[_ <: Suite], newInstance: () => Suite)
     if (PlatformCompat.isIgnoreSuite(cls)) {
       val description = getDescription()
       notifier.fireTestIgnored(description)
-      return Future.successful(())
-    }
-    var isBeforeAllRun = false
-    val result = {
-      val isContinue = runBeforeAll(notifier)
-      isBeforeAllRun = isContinue
-      if (isContinue) {
-        runAsyncTestsSynchronously(notifier)
-      } else {
-        Future.successful(())
+      Future.successful(())
+    } else {
+      val result =
+        for {
+          isBeforeAllRun <- runBeforeAll(notifier)
+          if isBeforeAllRun
+          _ <- runAsyncTestsSynchronously(notifier)
+        } yield isBeforeAllRun
+
+      result.transformWithCompat {
+        case util.Success(true) =>
+          runAfterAll(notifier)
+
+        case util.Success(false) =>
+          Future.successful(())
+
+        case util.Failure(ex) =>
+          Future.failed(ex)
       }
     }
-    result.transformCompat { s =>
-      if (isBeforeAllRun) {
-        runAfterAll(notifier)
-      }
-      s
-    }
   }
 
-  private def runBeforeAll(notifier: RunNotifier): Boolean = {
-    var isContinue = runHiddenTest(notifier, "beforeAll", suite.beforeAll())
-    suite.munitFixtures.foreach { fixture =>
-      isContinue &= runHiddenTest(
-        notifier,
-        s"beforeAllFixture(${fixture.fixtureName})",
-        fixture.beforeAll()
+  private def runBeforeAll(notifier: RunNotifier): Future[Boolean] = {
+    val suiteBeforeAllF = Future(
+      runHiddenTest(notifier, "beforeAll", suite.beforeAll())
+    )
+
+    def beforeAllFixtureF =
+      Future.sequence(
+        suite.munitFixtures.map(f =>
+          Future(
+            runHiddenTest(
+              notifier,
+              s"beforeAllFixture(${f.fixtureName})",
+              f.beforeAll()
+            )
+          )
+        )
       )
-    }
-    isContinue
+
+    def beforeAllAsyncFixtureF =
+      Future
+        .sequence(
```

Comment:
Same comment as before, we should avoid using `Future.sequence`

Reply:
Done.